server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2548,7 +2565,8 @@ public void compactAll(long compactionId, UserCompactionConfig compactionConfig)
     boolean updateMetadata = false;
 
     synchronized (this) {
-      if (lastCompactID >= compactionId)
+      // ACCUMULO-3645 compaction may generate results if there is a compaction-scope iterator
+      if (lastCompactID >= compactionId && compactionConfig.getIterators().isEmpty())
```

Comment:
If lastCompactID >= compactionId then should not do anything.  When master receives compaction request it increments counter in zookeeper.   When a tablet compacts, it records the compaction id in metadata.   Master keeps asking tablets to compact until all tablets are counter are >= the zookeeper value.   This change could lead to tablet compacting multiple times for a single user compaction request with iters.